src/plugins_types/union.c

```diff
@@ -150,6 +150,40 @@ lyb_parse_union(const void *lyb_data, size_t lyb_data_len, uint32_t *type_idx, c
     }
 }
 
+/**
+ * @brief For leafref failures, ensure the appropriate error is propagated, not a type validation failure.
```

Comment:
Please also mention the RFC 7950 section that defines the specific error and the error-app-tag.

Reply:
done